### PR TITLE
[AKS] `az aks create/update`: Add new parameter `--node-os-upgrade-channel` to specify which OS on your nodes is updated

### PR DESCRIPTION
**Pull request by paulgmiller**
…com/Azure/azure-cli-extensions/pull/5788/files

**Related command**
az aks create|update

**Description**
This allows the setting of node os upgrade channel in create and update of aks clusters.
Public docs are here. https://learn.microsoft.com/en-us/azure/aks/auto-upgrade-node-image

It is GAing this preview extension: https://github.com/Azure/azure-cli-extensions/pull/5788

This became available in the 06-01 api.

**Testing Guide**
Same tests as extension create and update with nodeos upgrade channel. Make sure its preserved. Combine with upgrade chennel

**History Notes**
[AKS] `az aks create/update`: Add new parameter `--node-os-upgrade-channel` to specify which OS on your nodes is updated

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [X] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [X] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [X] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Comment by paulgmiller** [comment]
@wangzelin007 and @FumingZhang  would you mind looking at this sicne you approved https://github.com/Azure/azure-cli-extensions/pull/5788 it should be a mirror of that.

Also we can remove code from preview extension after this is live. 

Since this gives customers a way to disable Unattended Upgrades (which caused a sev 1 across several regions on saturday we might actually ask for an expidtited release of this)

**Review by FumingZhang — APPROVED** [review]
LGTM

File: src/azure-cli/azure/cli/command_modules/acs/_consts.py [hunk 1]
```diff
@@ -86,6 +86,12 @@
 CONST_NODE_IMAGE_UPGRADE_CHANNEL = "node-image"
 CONST_NONE_UPGRADE_CHANNEL = "none"
 
+# consts for node os upgrade channel
+CONST_NODE_OS_CHANNEL_NODE_IMAGE = "NodeImage"
+CONST_NODE_OS_CHANNEL_NONE = "None"
+CONST_NODE_OS_CHANNEL_SECURITY_PATCH = "SecurityPatch"
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/_consts.py:92]: Seems the enum SecurityPatch is not defined in stable API, is there any specific reason?

https://github.com/Azure/azure-rest-api-specs/pull/24487

Comment (paulgmiller) [reply src/azure-cli/azure/cli/command_modules/acs/_consts.py:92]: Yeah that didn't ga let me rip it out.

**Comment by paulgmiller** [comment]
@zhoxing-ms  anythign else you need here?

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**